src/target/riscv/riscv.c

```diff
@@ -643,6 +643,73 @@ int riscv_remove_watchpoint(struct target *target,
 	return ERROR_OK;
 }
 
+int riscv_hit_watchpoint(struct target *target, struct watchpoint **hit_watchpoint)
```

Comment:
Can you add a brief comment of what this function is for, eg. "Sets *hit_watchpoint to the first trigger that was the cause for being halted."